### ISSUE TITLE
Need feature: the "changed" state

### ISSUE THREAD

**Issue by dzwiedziu-nkg** [joepuzzo/informed#26]
Currently you have `value` and `touched`. `value` is the current value (typed or achieved from `initialValues`). `touched` is set to `true` if input was focused and lost focus. I want to mark inputs who are changed value according to initial value. Typical application: edit form.

In `react-form` I can implement it in my custom input because I can achieve access to `defaultValues` via:
```javascript
contextTypes = {
  formProps: PropTypes.object
}
```

And `this.context.defaultValues['my_field_name']` have my initial value. So in my custom input I can compare value from `defaultValues` with value from field state and mark my custom input as changed.

In `informed` the `initialValues` is stored in controller but controller is used in `bindToField` and no propagated to `Field`. Moreover nor `withController` and not `FormContext` is not exported. So I have not access to controller from my custom field.

Can you implement access to `initialValue` from `formApi`/`fieldApi` or `formState`/`fieldState` or better: implement `changed` state next to `touched` state?

**Comment by joepuzzo** [comment]
" I want to mark inputs who are changed value according to initial value. " Can yo elaborate on this? im not following what you are trying to achieve here?

**Comment by joepuzzo** [comment]
You want some sort of flag that marks a field when it has been modified from its initial value?

**Comment by dzwiedziu-nkg** [comment]
Yes. Similar to `touched`. When `value` is different than `initialValue` it should be `true`. When `value` come back to equal to `initialValue` it should be `false` again.

**Comment by joepuzzo** [comment]
I dont think im going to add this. It is totally achievable for a developer to compare the current value that informed exposes to value that they had initially.. if you passed in initial value then you obviously have access to it. Im going to stick to my guns here and call this one out as a NOT 90% usecase senario. When i rewrote the lib I told myself that i was going to push back on things that were not needed to keep the lib slim and this is an example of one of those cases.

**Comment by AntowaKartowa** [comment]
@joepuzzo Have the same situation. To avoid saving not changed values and to disable save button when there is no changes at all. I am using custom field components and making comparison inside them setting touched to false or true inside onChange and onBlur handlers.

First I thought that only fields with touched set to true are added to `touched` object. But I have to change checking only the keys of the object to filter those which set to false. 

Then I found out that submiting the form makes all the fields set as touched. Then I decided to make a snapshot of `touched` object inside onChange handler assigned to form. But quickly I found out that onChange is called also during form submit flow. So I added hidden checkbox to each form to filter onChange events that happens during submit procedure. Yeah it is working. But for me it is clunky overloaded with destruction, reducers and other stuff solution. It would be better to have some mechanism to make same checks less complicated.
